binderhub/main.py

```diff
@@ -15,6 +15,7 @@ class MainHandler(BaseHandler):
     def get(self):
         self.render_template(
             "index.html",
+            badge_base_url=self.settings['badge_base_url'],
```

Comment:
`badge_base_url=self.settings['badge_base_url'] or f"{self.request.protocol}://{self.request.host}{self.settings['base_url']}",` -> maybe this is better. so you can remove `if else`s in js code and it becomes cleaner.

Reply:
I am not sure which one I prefer. I agree the JS has a few extra if statements, but I think I prefer it that way because it leaves the choice to the JS code. I dunno, it is just a feeling :-/